File: Builder/PHP/Reporter.php
<?php

require_once "IReportBuilder.php";

class Reporter {
    private IReportBuilder $builder;

    public function __construct(IReportBuilder $builder)
    {
        $this->builder = $builder;
    }

    public function createReport() {
        $this->builder->buildContent();
        $this->builder->buildFile();

        return $this->builder->getResult();
    }
}

File: Builder/Dart/Reporter.dart
import 'IReportBuilder.dart';
import 'Report.dart';

class Reporter {
  IReportBuilder? _builder;

  Reporter(this._builder);

  Report? getReport() {
    _builder?.buildContent();
    _builder?.buildFile();

    return _builder?.getResult();
  }
}
